lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -856,27 +857,41 @@ static Optional<ASTNode> findInnermostAncestor(
 static const Decl *findContainingDeclaration(SourceRange ReferenceRange,
                                              const DeclContext *ReferenceDC,
                                              const SourceManager &SM) {
-  if (const Decl *D = ReferenceDC->getInnermostDeclarationDeclContext())
+  auto ContainsReferenceRange = [&](const Decl *D) -> bool {
+    if (ReferenceRange.isInvalid())
```

Comment:
Could you remove this check from the `find_if` loop by moving the early return later in the function up?

Reply:
That changes behavior slightly, since in the top-level case it means we return `nullptr` and in the decl case we return the decl.

Reply:
Oh, you're right. I think you could still test the range's validity once before calling `find_if`, but that might not be worth the trouble.

Reply:
I guess I can remove the *other* check, even though that makes things slightly less efficient. What do you think?

Reply:
I don't think you should remove the other check. You should either put an `if (ReferenceRange.isInvalid()) return D;` before the `if (auto * IDC = ...)` block so it never calls `ContainsReferenceRange` if the reference range is invalid (and perhaps assert that fact in `ContainsReferenceRange`), or stick with this implementation.